### PR TITLE
Issue #565 - Initial draft of FHIR Validation Guide

### PR DESCRIPTION
**Pull request by JohnTimm**
Signed-off-by: John T.E. Timm <johntimm@us.ibm.com>

### CODE REVIEW

File: docs/src/pages/guides/FHIRValidationGuide.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+layout: post
+title: FHIR Validation Guide
+description: FHIR Validation Guide
+date:   2020-01-17 15:45:00 -0400
+permalink: /fhirvalidationguide/
+---
+
+## Overview
+
+The FHIR validation component (FHIRValidator) provides Java APIs for validating FHIR resources using constraints specified in their corresponding structure definitions. For example, in the Patient resource, we have the following constraint:
```
Comment (prb112) [inline docs/src/pages/guides/FHIRValidationGuide.md]:
```suggestion
The IBM FHIR Server Validation component (FHIRValidator) provides Java APIs for validating FHIR resources using constraints specified in their corresponding structure definitions. For example, in the Patient resource, we have the following constraint:
```

Comment (prb112) [reply docs/src/pages/guides/FHIRValidationGuide.md]: do we want to add the path to the component?  on bintray or in github?

File: docs/src/pages/guides/FHIRValidationGuide.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
+---
+layout: post
+title: FHIR Validation Guide
+description: FHIR Validation Guide
+date:   2020-01-17 15:45:00 -0400
+permalink: /fhirvalidationguide/
```
Comment (prb112) [inline docs/src/pages/guides/FHIRValidationGuide.md]:
```suggestion
permalink: /FHIRValidationGuide/
```
The link will end up being the above.

File: docs/src/pages/guides/FHIRValidationGuide.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+@Constraint(
+    id = "pat-1",
+    level = "Rule",
+    location = "Patient.contact",
+    description = "SHALL at least contain a contact's details or a reference to an organization",
+    expression = "name.exists() or telecom.exists() or address.exists() or organization.exists()"
+)
+```
+
+The validation component picks up this annotation, pulls out the FHIRPath expression and passes it on to the FHIRPath engine for evaluation. If the invariant evaluates to `false` then the FHIR validator will generate an OperationOutcome.Issue with the severity set relative to the "level" of the constraint (i.e. "Rule" or "Warning");
```
Comment (prb112) [inline docs/src/pages/guides/FHIRValidationGuide.md]:
```suggestion
The validation component picks up the Java annotation, pulls out the FHIRPath expression and passes it on to the FHIRPath engine for evaluation. If the invariant evaluates to `false` then the FHIR validator will generate an OperationOutcome.Issue with the severity set relative to the "level" of the constraint (i.e. "Rule" or "Warning");
```

File: docs/src/pages/guides/FHIRValidationGuide.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+
+## Profile Support
+
+The validation component will also validate a resource against profiles that it asserts conformance to in the `Resource.meta.profile` element assuming those profiles are available to the IBM FHIR Server via the FHIR registry component at runtime.
```
Comment (prb112) [inline docs/src/pages/guides/FHIRValidationGuide.md]:
```suggestion
The validation component will also validate a resource against profiles that it asserts conformance to in the `Resource.meta.profile` element assuming those profiles are available to the IBM FHIR Server via the FHIR registry component (fhir-registry) at runtime.
```

File: docs/src/pages/guides/FHIRValidationGuide.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    "max": "1",
+    ...
+    "type": [{
+        "code": "uri"
+    }],
+    "fixedUri": "http://loinc.org",
+     ...
+}
+```
+```
```
Comment (prb112) [inline docs/src/pages/guides/FHIRValidationGuide.md]:
```suggestion
``` json
```

Comment (prb112) [reply docs/src/pages/guides/FHIRValidationGuide.md]: adds nice md formatting in the generated site

File: docs/src/pages/guides/FHIRValidationGuide.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+- Reference type constraints (FHIRPath resolve/is/conformsTo functions)
+- Extension constraints (FHIRPath `conformsTo` function)
+- Vocabulary constraints (FHIRPath `memberOf` function)
+- Choice type constraints (FHIRPath `as` function)
+
+NOTE: there is currently no support for `closed` or `ordered` slices.
+
+For example, the HL7 bodyweight profile has the following cardinality and fixed value constraints:
+
+```
```
Comment (prb112) [inline docs/src/pages/guides/FHIRValidationGuide.md]:
```suggestion
``` json
```

File: docs/src/pages/guides/FHIRValidationGuide.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+which are used by the ConstraintGenerator to generate the following FHIRPath expression:
+
+```
+code.where(coding.where(system = 'http://loinc.org' and code = '29463-7').exists()).exists()
+```
+
+The HL7 bodyweight profile has the following reference type constraint:
+
+```
```
Comment (prb112) [inline docs/src/pages/guides/FHIRValidationGuide.md]:
```suggestion
``` json 
```

File: docs/src/pages/guides/FHIRValidationGuide.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    level = "Rule",
+    location = "Patient.contact",
+    description = "SHALL at least contain a contact's details or a reference to an organization",
+    expression = "name.exists() or telecom.exists() or address.exists() or organization.exists()"
+)
+```
+
+The validation component picks up this annotation, pulls out the FHIRPath expression and passes it on to the FHIRPath engine for evaluation. If the invariant evaluates to `false` then the FHIR validator will generate an OperationOutcome.Issue with the severity set relative to the "level" of the constraint (i.e. "Rule" or "Warning");
+
+
```
Comment (prb112) [inline docs/src/pages/guides/FHIRValidationGuide.md]:
```suggestion
![https://ibm.github.io/FHIR/images/fhir-dependency-graph.png](https://ibm.github.io/FHIR/images/fhir-dependency-graph.png)
```

Add the diagram here?

File: docs/src/pages/guides/FHIRValidationGuide.md (hunk 2 above)
Comment (JohnTimm) [reply docs/src/pages/guides/FHIRValidationGuide.md]: updated

File: docs/src/pages/guides/FHIRValidationGuide.md (hunk 3 above)
Comment (JohnTimm) [reply docs/src/pages/guides/FHIRValidationGuide.md]: updated

File: docs/src/pages/guides/FHIRValidationGuide.md (hunk 2 above)
Comment (JohnTimm) [reply docs/src/pages/guides/FHIRValidationGuide.md]: updated

File: docs/src/pages/guides/FHIRValidationGuide.md (hunk 3 above)
Comment (JohnTimm) [reply docs/src/pages/guides/FHIRValidationGuide.md]: updated

File: docs/src/pages/guides/FHIRValidationGuide.md (hunk 4 above)
Comment (JohnTimm) [reply docs/src/pages/guides/FHIRValidationGuide.md]: updated

File: docs/src/pages/guides/FHIRValidationGuide.md (hunk 5 above)
Comment (JohnTimm) [reply docs/src/pages/guides/FHIRValidationGuide.md]: updated

File: docs/src/pages/guides/FHIRValidationGuide.md (hunk 6 above)
Comment (JohnTimm) [reply docs/src/pages/guides/FHIRValidationGuide.md]: updated

File: docs/src/pages/guides/FHIRValidationGuide.md (hunk 7 above)
Comment (JohnTimm) [reply docs/src/pages/guides/FHIRValidationGuide.md]: updated

File: docs/src/pages/guides/FHIRValidationGuide.md (hunk 8 above)
Comment (JohnTimm) [reply docs/src/pages/guides/FHIRValidationGuide.md]: updated

File: docs/src/pages/guides/FHIRValidationGuide.md (hunk 1 above)
Comment (JohnTimm) [reply docs/src/pages/guides/FHIRValidationGuide.md]: added links to the components referenced

**Pull request merged**